sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java

```diff
@@ -45,27 +46,33 @@
     static HttpPipeline buildPipeline(
         TablesSharedKeyCredential tablesSharedKeyCredential,
         TokenCredential tokenCredential, AzureSasCredential azureSasCredential, String sasToken,
-        String endpoint, RequestRetryOptions retryOptions, HttpLogOptions logOptions,
-        HttpClient httpClient, List<HttpPipelinePolicy> additionalPolicies,
-        Configuration configuration, ClientLogger logger) {
+        String endpoint, RequestRetryOptions retryOptions, HttpLogOptions logOptions, ClientOptions clientOptions,
+        HttpClient httpClient, List<HttpPipelinePolicy> perCallAdditionalPolicies,
+        List<HttpPipelinePolicy> perRetryAdditionalPolicies, Configuration configuration, ClientLogger logger) {
+
+        configuration = (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
+        clientOptions = (clientOptions == null) ? new ClientOptions() : clientOptions;
+        logOptions = (logOptions == null) ? new HttpLogOptions() : logOptions;
```

Comment:
@alzimmermsft If we use the default constructor for `HttpLogOptions` it will use the default headers allow-list, as opposed to passing null to `HttpLoggingPolicy`, what is the desired default behavior?

Reply:
The default `HttpLogOptions` constructor uses `HttpLogDetailLevel.NONE` which results in no logging occurring.